rust/lancedb/src/query.rs

```diff
@@ -71,6 +73,7 @@ impl Query {
             use_index: true,
             filter: None,
             select: None,
+            select_with_transform: None,
```

Comment:
Minor nit: if the word "transform" is already used elsewhere then maybe this ship has sailed but I'd prefer this to be called `select_with_projection` if possible.